core/vm/contracts.go

```diff
@@ -132,3 +145,190 @@ func (c *dataCopy) RequiredGas(input []byte) uint64 {
 func (c *dataCopy) Run(in []byte) ([]byte, error) {
 	return in, nil
 }
+
+// bigModExp implements a native big integer exponential modular operation.
+type bigModExp struct{}
+
+// RequiredGas returns the gas required to execute the pre-compiled contract.
+func (c *bigModExp) RequiredGas(input []byte) uint64 {
+	// Pad the input with zeroes to the minimum size to read the field lengths
+	input = common.RightPadBytes(input, 96)
+
+	var (
+		baseLen = new(big.Int).SetBytes(input[:32])
+		expLen  = new(big.Int).SetBytes(input[32:64])
+		modLen  = new(big.Int).SetBytes(input[64:96])
+	)
+	input = input[96:]
+
+	// Retrieve the head 32 bytes of exp for the adjusted exponent length
+	var expHead *big.Int
+	if big.NewInt(int64(len(input))).Cmp(baseLen) <= 0 {
+		expHead = new(big.Int)
+	} else {
+		offset := int(baseLen.Uint64())
+
+		input = common.RightPadBytes(input, offset+32)
+		if expLen.Cmp(big.NewInt(32)) > 0 {
+			expHead = new(big.Int).SetBytes(input[offset : offset+32])
+		} else {
+			expHead = new(big.Int).SetBytes(input[offset : offset+int(expLen.Uint64())])
+		}
+	}
+	// Calculate the adjusted exponent length
+	var msb int
+	if bitlen := expHead.BitLen(); bitlen > 0 {
+		msb = bitlen - 1
+	}
+	adjExpLen := new(big.Int)
+	if expLen.Cmp(big.NewInt(32)) > 0 {
+		adjExpLen.Sub(expLen, big.NewInt(32))
+		adjExpLen.Mul(big.NewInt(8), adjExpLen)
+	}
+	adjExpLen.Add(adjExpLen, big.NewInt(int64(msb)))
+
+	// Calculate the gas cost of the operation
+	gas := new(big.Int).Set(math.BigMax(modLen, baseLen))
+	switch {
+	case gas.Cmp(big.NewInt(64)) <= 0:
+		gas.Mul(gas, gas)
+	case gas.Cmp(big.NewInt(1024)) <= 0:
+		gas = new(big.Int).Add(
+			new(big.Int).Div(new(big.Int).Mul(gas, gas), big.NewInt(4)),
+			new(big.Int).Sub(new(big.Int).Mul(big.NewInt(96), gas), big.NewInt(3072)),
+		)
+	default:
+		gas = new(big.Int).Add(
+			new(big.Int).Div(new(big.Int).Mul(gas, gas), big.NewInt(16)),
+			new(big.Int).Sub(new(big.Int).Mul(big.NewInt(480), gas), big.NewInt(199680)),
+		)
+	}
+	gas.Mul(gas, math.BigMax(adjExpLen, big.NewInt(1)))
+	gas.Div(gas, big.NewInt(100))
```

Comment:
Where does it state 100 ? 

https://github.com/ethereum/EIPs/blob/4d4d8fb75eedeacc6456b804d9fb58c1778f1c76/EIPS/bigint_modexp.md#specification says 

> Consumes `floor(mult_complexity(max(length_of_MODULUS, length_of_BASE)) * max(ADJUSTED_EXPONENT_LENGTH, 1) / GQUADDIVISOR)` gas